File: 06_Algorithm/leetcode_342.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
# File              : leetcode_342.py
# Author            : Andy
# Date              : 2022.08.29
# Last Modified Date: 2022.08.29
# Last Modified By  : Andy


class Solution:

    def isPowerOfFour(self, n: int) -> bool:
        for i in range(16):
            if n == 4**i:
                return True
        return False

File: Algorithm/leetcode_342.go
/**
 * File              : leetcode_342.go
 * Author            : Andy
 * Date              : 2022.08.29
 * Last Modified Date: 2022.08.29
 * Last Modified By  : Andy
 */

package isPowerOfFour

import "math"

func isPowerOfFour(n int) bool {
	for i := 0; i < 16; i++ {
		if n == int(math.Pow(4, float64(i))) {
			return true
		}
	}
	return false
}
